File: 270-closest-binary-search-tree-value/270-closest-binary-search-tree-value.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def closestValue(self, root, target):
        """
        :type root: TreeNode
        :type target: float
        :rtype: int
        """
        currNode, closest = root, root.val
        while (currNode is not None):
            if abs(currNode.val-target) < abs(closest-target):
                closest = currNode.val
            if currNode.val < target:
                currNode = currNode.right
            elif currNode.val > target:
                currNode = currNode.left
            else:
                break
        return closest
        

File: 270-closest-binary-search-tree-value/270-closest-binary-search-tree-value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int closestValue(TreeNode root, double target) {
        TreeNode currNode = root;
        int closest = root.val;
        while (currNode != null) {
            if (Math.abs(currNode.val-target) < Math.abs(closest-target)) {
                closest = currNode.val;
            }
            if (currNode.val < target) {
                currNode = currNode.right;
            }
            else if (currNode.val > target) {
                currNode = currNode.left;
            }
            else {
                break;
            }
        }
        return closest;
    }
}